mautic/mautic
# Fix for cke5 to allow link attributes

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #13724 



#### Description:

This partly reenables a feature that was present in Mautic 4: allowing for more html and html properties (attributes) to be printed inside CKE.  
This change is limited to:  
- attributes on Links
- allowing the use of `span` tags, with attributes _(used by some email template makers for special highlights in placeholder text or even to override link colors)_ 
 
I also had to upgrade CKE (went from v38 to 41 ), because there was a bug that gave duplicate module errors when trying to import and add certain plugins.

When editing the contact block in the Brienz theme (and similar issues will arise in people's custom themes), the CKEditor gave a blank textarea, instead of letting you edit the existing text.
This was caused by some html attributes being in the prefilled Text Block. Instead of just the attributes being filtered out, the entire content disappeared.
![Screenshot 2024-05-06 at 13 25 40](https://github.com/mautic/mautic/assets/758957/093cc524-b731-4dca-adbf-757e1a899f7f)
![Screenshot 2024-05-06 at 13 25 57](https://github.com/mautic/mautic/assets/758957/a96cda20-9e69-48e3-9dd2-c858776d26ba) 

[Read the full bug report here](https://github.com/mautic/mautic/issues/13724)




#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a new emails
3. Use the Brienz theme
4. Open the contact block
5. The content in CKE is not empty
6. Add a bit of text, and save it
7. Your preview shows the changed text
8. Extra: if you inspect the Block in the Browser (using dev tools), you should see that the links (eg. for phone and email) still contain a title attribute (we don't really care for the ID or the 'draggable' attributes as they are set by the system for interaction and not by the user).
9. Test that the other functionalities of CKE still work, eg. in another Text block: make some bold text, make a list, a link, adding a Token, …
